documentapi/src/vespa/documentapi/messagebus/policies/contentpolicy.h

```diff
@@ -6,55 +6,62 @@
 #include <vespa/vdslib/distribution/distribution.h>
 #include <vespa/document/bucket/bucketidfactory.h>
 #include <vespa/messagebus/routing/hop.h>
+#include <mutex>
 
 namespace config {
     class ICallback;
     class ConfigFetcher;
 }
 
-namespace storage {
-namespace lib {
+namespace storage::lib {
     class Distribution;
     class ClusterState;
 }
-}
 
 namespace documentapi {
 
 class ContentPolicy : public ExternSlobrokPolicy
 {
 private:
-    document::BucketIdFactory _bucketIdFactory;
-    std::unique_ptr<storage::lib::ClusterState> _state;
-    string _clusterName;
-    string _clusterConfigId;
-    std::unique_ptr<config::ICallback>          _callBack;
-    std::unique_ptr<config::ConfigFetcher>      _configFetcher;
-    std::unique_ptr<storage::lib::Distribution> _distribution;
-    std::unique_ptr<storage::lib::Distribution> _nextDistribution;
+    document::BucketIdFactory                         _bucketIdFactory;
+    mutable std::mutex                                _lock;
```

Comment:
Could this be a read/write lock ?

Reply:
The critical section in the expected code path only spans two atomic up-refs, but write-locking should be very rare so it might be beneficial anyway. Will replace with `shared_mutex`.